ee/maintained-apps/inputs/darwin/ingester.go

```diff
@@ -0,0 +1,370 @@
+			Refs:     scripts,
+		}
+
+		outBytes, err := json.MarshalIndent(outFile, "", "  ")
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "marshaling output app manifest")
+		}
+
+		// Overwrite the file, since right now we're only caring about 1 version (latest). If we
+		// care about previous data, it will be in our Git history.
+		if err := os.WriteFile(path.Join(outputPath, string(fleet.MacOSPlatform), fmt.Sprintf("%s.json", input.SourceIdentifier)), outBytes, 0o644); err != nil {
+			return ctxerr.Wrap(ctx, err, "writing output json file")
+		}
+
+		if err := updateAppsListFile(ctx, input, outApp); err != nil {
+			return ctxerr.Wrap(ctx, err, "updating apps list file")
+		}
+
+	}
+
+	return nil
+}
+
+const baseBrewAPIURL = "https://formulae.brew.sh/api/"
+
+type brewIngester struct {
+	baseURL string
+	logger  kitlog.Logger
+	client  *http.Client
+}
+
+func (i *brewIngester) IngestOne(ctx context.Context, app maintained_apps.InputApp) (*maintained_apps.FMAManifestApp, map[string]string, error) {
+	apiURL := fmt.Sprintf("%scask/%s.json", i.baseURL, app.SourceIdentifier)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, apiURL, nil)
+	if err != nil {
+		return nil, nil, ctxerr.Wrap(ctx, err, "create http request")
+	}
+
+	res, err := i.client.Do(req)
+	if err != nil {
+		return nil, nil, ctxerr.Wrap(ctx, err, "execute http request")
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, nil, ctxerr.Wrap(ctx, err, "read http response body")
+	}
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		// success, go on
+	case http.StatusNotFound:
+		// nothing to do
+		level.Warn(i.logger).Log("msg", "maintained app missing in brew API", "identifier", app.SourceIdentifier)
+		return nil, nil, nil
+	default:
+		if len(body) > 512 {
+			body = body[:512]
+		}
+		return nil, nil, ctxerr.Errorf(ctx, "brew API returned status %d: %s", res.StatusCode, string(body))
+	}
+
+	var cask brewCask
+	if err := json.Unmarshal(body, &cask); err != nil {
+		return nil, nil, ctxerr.Wrapf(ctx, err, "unmarshal brew cask for %s", app.SourceIdentifier)
+	}
+
+	out := &maintained_apps.FMAManifestApp{}
+
+	// validate required fields
+	if len(cask.Name) == 0 || cask.Name[0] == "" {
+		return nil, nil, ctxerr.Errorf(ctx, "missing name for cask %s", app.SourceIdentifier)
+	}
+	if cask.Token == "" {
+		return nil, nil, ctxerr.Errorf(ctx, "missing token for cask %s", app.SourceIdentifier)
+	}
+	if cask.Version == "" {
+		return nil, nil, ctxerr.Errorf(ctx, "missing version for cask %s", app.SourceIdentifier)
+	}
+	if cask.URL == "" {
+		return nil, nil, ctxerr.Errorf(ctx, "missing URL for cask %s", app.SourceIdentifier)
+	}
+	_, err = url.Parse(cask.URL)
+	if err != nil {
+		return nil, nil, ctxerr.Wrapf(ctx, err, "parse URL for cask %s", app.SourceIdentifier)
+	}
+
+	out.Version = cask.Version
+	out.InstallerURL = cask.URL
+	out.UniqueIdentifier = app.UniqueIdentifier
+	out.SHA256 = cask.SHA256
```

Comment:
There's a nocheck thing for Chrome right?

Reply:
Yes. I think that since I'm doing a deeper diff now on the existing vs new outputs, we should be good here.